monai/transforms/croppad/array.py

```diff
@@ -802,16 +804,18 @@ def crop_pad(
         slices = self.compute_slices(roi_start=box_start, roi_end=box_end)
```

Comment:
If `margin>0`, `roi_start` maybe < 0, but `slices` can only start from 0, then `cropped.peek_pending_shape()` would be the wrong shape.

Reply:
maybe after line 814 padding with -box_start, the shape will be correct again?

Reply:
No, since line 811-813 was using `cropped.peek_pending_shape()` to compute pad width.

Reply:
I find the reason, line 818 only updated "pad_info" in `TraceKeys.EXTRA_INFO` for padding transform which may lose `lazy_shape` and `lazy_affine`.

Reply:
solved when we updated line 818.

Reply:
> Hi @wyli , @KumoLiu
> Today I got stuck in a test case because of `set_track_meta`. Do you think if we should add a check of `_TRACK_META` when doing lazy resampling? If False, maybe we can raise a warning to let users know that they cannot get expected results.

Yes, if False, we don't update the metatensor's meta info and lazy resampling is not available.